api/org/commcare/api/json/PromptToJson.java

```diff
@@ -0,0 +1,227 @@
+                obj.put("type", "sub-group");
+                obj.put("repeatable", false);
+                break;
+            case FormEntryController.EVENT_REPEAT:
+                // we're in a subgroup
+                parseCaption(model.getCaptionPrompt(), obj);
+                obj.put("type", "sub-group");
+                obj.put("repeatable", true);
+                obj.put("exists", true);
+                break;
+            case FormEntryController.EVENT_PROMPT_NEW_REPEAT:
+                // we're in a subgroup
+                parseCaption(model.getCaptionPrompt(), obj);
+                obj.put("type", "sub-group");
+                obj.put("repeatable", true);
+                obj.put("exists", false);
+                break;
+        }
+    }
+
+    private static void parseRepeatJuncture(FormEntryModel model, JSONObject obj, FormIndex ix) {
+        FormEntryCaption formEntryCaption = model.getCaptionPrompt(ix);
+        FormEntryCaption.RepeatOptions repeatOptions = formEntryCaption.getRepeatOptions();
+        parseCaption(formEntryCaption, obj);
+        obj.put("header", repeatOptions.header);
+        obj.put("repetitions", formEntryCaption.getRepetitionsText());
+        obj.put("add-choice", repeatOptions.add);
+        obj.put("delete-choice", repeatOptions.delete);
+        obj.put("del-header", repeatOptions.delete_header);
+        obj.put("done-choice", repeatOptions.done);
+    }
+
+    private static void parseQuestionAnswer(JSONObject obj, FormEntryPrompt prompt){
+        IAnswerData answerValue = prompt.getAnswerValue();
+        if (answerValue == null){
+            obj.put("answer", JSONObject.NULL);
+            return;
+        }
+        switch(prompt.getDataType()) {
+            case Constants.DATATYPE_NULL:
+            case Constants.DATATYPE_TEXT:
+                obj.put("answer", answerValue.getDisplayText());
+                return;
+            case Constants.DATATYPE_INTEGER:
+                obj.put("answer", (int)answerValue.getValue());
+                return;
+            case Constants.DATATYPE_LONG:
+            case Constants.DATATYPE_DECIMAL:
+                obj.put("answer", (double)answerValue.getValue());
+                return;
+            case Constants.DATATYPE_DATE:
+                obj.put("answer", (DateUtils.formatDate((Date)answerValue.getValue(), DateUtils.FORMAT_ISO8601)));
+                return;
+            case Constants.DATATYPE_TIME:
+                obj.put("answer", answerValue.getDisplayText());
+                return;
+            case Constants.DATATYPE_DATE_TIME:
+                obj.put("answer", ((Date)answerValue.getValue()).getTime()/1000);
+                return;
+            case Constants.DATATYPE_CHOICE:
+                Selection singleSelection = ((Selection)answerValue.getValue());
+                singleSelection.attachChoice(prompt.getQuestion());
+                obj.put("answer", ((Selection)answerValue.getValue()).index + 1);
```

Comment:
Can you internalize the `((Selection)answerValue.getValue()).index + 1` into the `Selection` class so that if we change how indexing works we don't have to change the `+ 1` here?
Something like adding `Selection.getAnswerIndex()`